### PR TITLE
feat(operator): helm deployment initialization

### PR DESCRIPTION
**Pull request by louhde**

### CODE REVIEW

File: components/operator/config/rbac/role.yaml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 ---
 apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRole
```
Comment (gfyrag) [inline components/operator/config/rbac/role.yaml:3]: Are you sure it works?
The operator use some cluster scoped resources.

Comment (louhde) [reply components/operator/config/rbac/role.yaml:3]: Didn't sure .. I take a look why Kubebuilder changed it

File: components/operator/config/rbac/role.yaml [hunk 2]
```diff
@@ -132,3 +132,15 @@ rules:
   - get
   - list
   - watch
+- apiGroups:
```
Comment (gfyrag) [inline components/operator/config/rbac/role.yaml:135]: Can be removed (not a blocker).

Comment (louhde) [reply components/operator/config/rbac/role.yaml:135]: I prefer to keep it I think I will need it in the future

File: components/operator/Makefile [hunk 3]
```diff
@@ -102,6 +102,37 @@ deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in
 undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
 	$(KUSTOMIZE) build config/default | kubectl delete --ignore-not-found=$(ignore-not-found) -f -
 
+.PHONY: helm-update
+helm-update: manifests generate
```
Comment (flemzord) [inline components/operator/Makefile:106]: Maybe need to exclude Kutomiszation.yaml ?

File: components/operator/internal/controllers/stack/stack_reconciler.go [hunk 4]
```diff
@@ -42,6 +42,7 @@ const (
 // +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=core,resources=namespaces,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=cert-manager.io,resources=certificates,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=traefik.containo.us,resources=middlewares,verbs=get;list;watch;create;update;patch;delete
```
Comment (flemzord) [inline components/operator/internal/controllers/stack/stack_reconciler.go]: Remove this line because i don't need this.

Comment (louhde) [reply components/operator/internal/controllers/stack/stack_reconciler.go]: Done

File: components/operator/Makefile (hunk 3 above)
Comment (louhde) [reply components/operator/Makefile:106]: I did it in .helmignore and .gitignore

**Pull request merged**